spec/integration/client_spec.rb

```diff
@@ -263,4 +263,29 @@ def terminate_workflow(input)
       expect(interceptor.results[5]).to eq([:terminate_workflow, handle.id])
     end
   end
+
+  # This case ensures we handle thread interruption (such as Thread#raise) properly
+  describe 'plays nice with GVL' do
```

Comment:
When you get to activity tests, I would like to see a (platform specific) test that uses `Kernel#exec` to start a sleep for several seconds and confirm you can cancel it.